### PR TITLE
Task1-Model

### PR DESCRIPTION
**Pull request by Oleksandr-Mokryi**
base model realization

### CODE REVIEW

File: DeliveryApp/DeliveryApp/Enums/ProductCategory.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DeliveryApp
```
Comment (yevhenbilushko) [inline DeliveryApp/DeliveryApp/Enums/ProductCategory.cs:7]: please fix spacing and formating

File: DeliveryApp/DeliveryApp/Enums/ProductCategory.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+{
+        public enum ProductCategory
+    {
+            None,
+            Pizza,
+            Drinks,
+            Fruits,
+            Salads,
+            Meet
```
Comment (yevhenbilushko) [inline DeliveryApp/DeliveryApp/Enums/ProductCategory.cs:16]: seems like "Meat"

File: DeliveryApp/DeliveryApp/Units/Base.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+﻿using System.Collections.ObjectModel;
+
+namespace DeliveryApp
+{
+    public class Base:IBase
```
Comment (yevhenbilushko) [inline DeliveryApp/DeliveryApp/Units/Base.cs:5]: no need "Base" and "IBase"
u wiil create EF Context for it in the future

File: DeliveryApp/DeliveryApp/IFaces/IUnit.cs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+﻿namespace DeliveryApp
+{
+    public interface IUnit
```
Comment (yevhenbilushko) [inline DeliveryApp/DeliveryApp/IFaces/IUnit.cs:3]: IUnit - not clear name
maybe User ?
please remove interfaces 
this task is for DB structure

File: DeliveryApp/DeliveryApp/IFaces/IProductWorker.cs [hunk 5]
```diff
@@ -0,0 +1,10 @@
+﻿using System.Collections.Generic;
+
+namespace DeliveryApp
+{
+    public interface IProductWorker
```
Comment (yevhenbilushko) [inline DeliveryApp/DeliveryApp/IFaces/IProductWorker.cs:5]: IProductWorker - sseems like it is for logic
this task is for DB structure
 remove it please

File: DeliveryApp/DeliveryApp/IFaces/IVendorWorker.cs [hunk 6]
```diff
@@ -0,0 +1,10 @@
+﻿using System.Collections.Generic;
+
+namespace DeliveryApp
+{
+    public interface IVendorWorker
```
Comment (yevhenbilushko) [inline DeliveryApp/DeliveryApp/IFaces/IVendorWorker.cs:5]: remove it

File: DeliveryApp/DeliveryApp/Units/Delivery.cs [hunk 7]
```diff
@@ -0,0 +1,13 @@
+﻿namespace DeliveryApp
+{
+    public class Delivery : IUnit
```
Comment (yevhenbilushko) [inline DeliveryApp/DeliveryApp/Units/Delivery.cs:3]: seems like "FirstName " "SecondName" no need for delivery
or what do u mean?
maybe Deliveryman?

File: DeliveryApp/DeliveryApp/Units/Order.cs [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+        public decimal Price { get; set; }
+        public OrderStatus Status { get; set; }
+        public Delivery Service { get; set; }
+
+        public Order()
+        {
+            OrderedProducts = new Dictionary<Product, int>();
+        }
+    }
+    public enum OrderStatus
```
Comment (yevhenbilushko) [inline DeliveryApp/DeliveryApp/Units/Order.cs:23]: move to separate file

File: DeliveryApp/DeliveryApp/Units/Product.cs [hunk 9]
```diff
@@ -0,0 +1,12 @@
+﻿namespace DeliveryApp
+{
+    public class Product
+    {
+        public string Name { get; set; }
+        public string Description { get; set; }
+        public decimal Price { get; set; }
+        public Vendor Vendor { get; init; }
```
Comment (yevhenbilushko) [inline DeliveryApp/DeliveryApp/Units/Product.cs:8]: add VendorId

File: DeliveryApp/DeliveryApp/Units/Product.cs [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+        public ProductCategory Category { get; set; }
```
Comment (yevhenbilushko) [inline DeliveryApp/DeliveryApp/Units/Product.cs:9]: add CategoryId

File: DeliveryApp/DeliveryApp/Units/Vendor.cs [hunk 11]
```diff
@@ -0,0 +1,12 @@
+﻿namespace DeliveryApp
+{
+    public class Vendor : IUnit
+    {
+        public string FirstName { get; set; }
+        public string SecondName { get; set; }
+        public string Address { get; set; }
+        public string Email { get; set; }
+        public string PhoneNumber { get; set; }
+        public VendorType Type { get; set; }
```
Comment (yevhenbilushko) [inline DeliveryApp/DeliveryApp/Units/Vendor.cs:10]: add TypeId

File: DeliveryApp/DeliveryApp/IFaces/IOrderWorker.cs [hunk 12]
```diff
@@ -0,0 +1,18 @@
+﻿using System.Collections.Generic;
+
+namespace DeliveryApp
+{
+    public interface IOrderWorker
```
Comment (yevhenbilushko) [inline DeliveryApp/DeliveryApp/IFaces/IOrderWorker.cs:5]: remove it

File: DeliveryApp/DeliveryApp/IFaces/ICustomerWorker.cs [hunk 13]
```diff
@@ -0,0 +1,10 @@
+﻿using System.Collections.Generic;
+
+namespace DeliveryApp
+{
+    public interface ICustomerWorker
```
Comment (yevhenbilushko) [inline DeliveryApp/DeliveryApp/IFaces/ICustomerWorker.cs:5]: remove it

File: DeliveryApp/DeliveryApp/Units/Product.cs (hunk 10 above)
Comment (Oleksandr-Mokryi) [reply DeliveryApp/DeliveryApp/Units/Product.cs:9]: In future tasks i created separate classes for all enums. And class has id field

File: DeliveryApp/DeliveryApp/Units/Base.cs (hunk 3 above)
Comment (Oleksandr-Mokryi) [reply DeliveryApp/DeliveryApp/Units/Base.cs:5]: I think i need Base for Linq requests. I need one place to save added objects

File: DeliveryApp/DeliveryApp/Enums/ProductCategory.cs [hunk 14]
```diff
@@ -0,0 +1,12 @@
+﻿namespace DeliveryApp.Enums
+{
+        public enum ProductCategory
```
Comment (yevhenbilushko) [inline DeliveryApp/DeliveryApp/Enums/ProductCategory.cs:3]: fix spacing before "public" word

File: DeliveryApp/DeliveryApp/Units/Order.cs [hunk 15]
```diff
@@ -0,0 +1,19 @@
+﻿using DeliveryApp.Enums;
+using System;
+using System.Collections.Generic;
+
+namespace DeliveryApp.Units
+{
+    public class Order
+    {
+        public Customer Customer { get; init; }
+        public IDictionary<Product, int> OrderedProducts { get; set; }
```
Comment (yevhenbilushko) [inline DeliveryApp/DeliveryApp/Units/Order.cs:10]: better to make just List of Products

**Review by yevhenbilushko — APPROVED** [review]

**Review by yevhenbilushko — APPROVED** [review]

**Pull request merged**